### PR TITLE
22182 cigarette pack is too big inside

### PR DESCRIPTION
**Pull request by iztokbajcar**
## About the PR

I made cigarettes the only allowed item inside cigarette packs.

## Why / Balance

It was done to prevent players from abusing the cig packs as a storage space for things that don't actually belong there.
Related issue: #22182

## Technical details

In `Resources/Prototypes/Entities/Objects/Consumables/Smokeables/Cigarettes/packs.yaml`, the `Cigarette` tag was whitelisted in the Storage components of the `CigPackBase` and `CigPackMixedBase` entities.

## Media

Before (a chocolate bar can be seen in the packet's inventory):
Screenshot_20240425_120007

After (I cannot place a chocolate bar inside):
Screenshot_20240425_121256

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- tweak: Cigarette packs can now only store cigarettes.

### CODE REVIEW

**Comment by ElPidor** [comment]
It is necessary to find a new way to store ammunition, and not completely destroy this method, since the use of a cigarette pack is effective due to its size and complexity of use, given the lack of alternatives. It may be worth increasing the weight of the cartridges or finding another storage method that does not lead to “abuse”
in 2 words
Skill issue

**Comment by Hanzdegloker** [comment]
Damn, I really liked putting my lighter in the thing for safe keeping

**Review by keronshb — CHANGES_REQUESTED** [review]
Cig Packs are designed to allow more than just cigarettes. If you had some bullets or pills it's completely fine to put into a cigarette pack for safe keeping.

**Comment by BellwetherLogic** [comment]
Considering that cigs are functionally cosmetic anyway (except for Interdyne Herbals) it feels kinda weird to disallow a small container from holding a lot of little things that aren't also cigs. If you're going to add a whitelist to it I think anything that's tiny and cylindrical should be able to take advantage of the TARDIS-like nature of the humble cig pack... which includes bullets anyway. 

Now, if you wanted to exclude things like syringes, flares, breath masks, etc, I could understand that. (Please don't, I like having somewhere to put all the 1x1 shit that comes in my survival box and makes storage a pain to manage.) But at the very least, lighters should absolutely be whitelisted for this even if an only-smoking-related whitelist gets applied to cig packs.

**Comment by K-Dynamic** [comment]
Or maybe make cigarette packs act like ammo and swab dispensers?

**Comment by Hanzdegloker** [comment]
> Or maybe make cigarette packs act like ammo and swab dispensers?

Absolutely not, I want to be able to PICK which cigs I smoke. Dan's soaked smokes foreva

also custom made cigs with like meth from chem

**Comment by ps3moira** [comment]
I think it'd be fine to allow things cigarette shaped to fit inside. Like a pen or crayon.

File: Resources/Prototypes/Entities/Objects/Consumable/Smokeables/Cigarettes/packs.yml [hunk 1]
```diff
@@ -49,6 +49,9 @@
   - type: Storage
     grid:
     - 0,0,4,1
+    whitelist:
+      tags:
+        - Cigarette
```
Comment (ike709) [inline Resources/Prototypes/Entities/Objects/Consumable/Smokeables/Cigarettes/packs.yml:54]: They're supposed to hold lighters as well.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Agree with keron here.

If you're going to add a big whitelist to it what's the point of the whitelist...?

**Comment by EmoGarbage404** [comment]
Closed due to feature freeze May 10th-June 14th. Comment to have it reopen after this.

**Pull request closed without merging**